src/Components/Components/src/PersistentComponentState.cs

```diff
@@ -16,6 +16,7 @@ public class PersistentComponentState
     private readonly IDictionary<string, byte[]> _currentState;
 
     private readonly List<PersistComponentStateRegistration> _registeredCallbacks;
+    private readonly List<(IPersistentComponentStateScenario Scenario, Action Callback, bool IsRecurring)> _restoringCallbacks = new();
```

Comment:
Do not use a tuple, use a proper `readonly struct RestoreComponentStateRegistration`